File: hw/toscano/u3/hw30Strategy_Fixed/src/main/java/ec/edu/espe/hw30_strategy/controller/BubbleSort.java
package ec.edu.espe.hw30_strategy.controller;

/**
 * BubbleSort algorithm as a strategy
 * @author Damian Toscano
 */
public class BubbleSort implements SortingStrategy {
    @Override
    public int[] sort(int data[]) {
        int n = data.length;
        int temp = 0;
        for(int i=0; i < n; i++){
            for(int j=1; j < (n-i); j++){
                if(data[j-1] > data[j]){
                    //swap elements
                    temp = data[j-1];
                    data[j-1] = data[j];
                    data[j] = temp;
                }
            }
        }
        return data;
    }
}

File: exams/u3/saraguro/EXAMEN_3P_Strategy_Pattern/SortingApp/src/strategies/BubbleSort.js
class BubbleSort {
    sort(array) {
        let len = array.length;
        for (let i = 0; i < len; i++) {
            for (let j = 0; j < len - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    let temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
        return array;
    }
}

module.exports = BubbleSort;
